### PR TITLE
Basic app behaviour and test helper

### PR DESCRIPTION
**Pull request by danielherrerohernando**
## What’s the focus of this PR
This PR adds basic greeting functions along with their unit tests.
Besides that, it adds integration/e2e tests to check real cli output, being the most important thing the way we manage the child process spawned by the test runner. This way we can mock certain functions (like testing different hours for welcome message) or test different inputs while keeping an easy asynchronous test flow. Further development will come to accomplish this.

## How to review this PR
Check basic functions and especially the testHelper file which involves the good stuff around testing the cli app.

Before submitting this PR, I made sure:

- [x] The code builds clean without any errors or warnings

### CODE REVIEW

File: test/integration-e2e/ohceCliApp.test.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
+const { expect } = require('chai');
+const testHelper = require('../testHelper/testHelper');
+
 describe('Integration-E2E Tests', () => {
 	describe('The ohce cli app', () => {
-		it('Should do sth', () => {});
+		it('Should greet user on startup', async () => {
+			const output = await testHelper.execute('./src/index.js', 'Daniel');
```
Comment (Betisman) [inline test/integration-e2e/ohceCliApp.test.js]:
```suggestion
      const greet = testHelper.execute('./src/index.js'); // so we can reuse the function
			const output = await greet('Daniel');
```

File: test/testHelper/testHelper.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { spawn } = require('child_process');
+const concat = require('concat-stream');
+
+const execute = (processPath, userName) => {
```
Comment (Betisman) [inline test/testHelper/testHelper.js]: Let's curry!!
```suggestion
const execute = (processPath) => (userName) => {
```

File: test/testHelper/testHelper.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	const childProcess = spawn('node', [processPath, userName]);
+	const promise = new Promise((resolve, reject) => {
+		childProcess.stdout.setEncoding('utf-8');
+		childProcess.on('error', reject);
+		childProcess.stderr.once('data', err => reject(err));
+		childProcess.stdout.pipe(concat(result => resolve(result)));
```
Comment (Betisman) [inline test/testHelper/testHelper.js]: loving that pipe.

File: test/unit/greetings.test.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const { expect } = require('chai');
+const { greetUser, goodbyeUser } = require('../../src/greetings');
+
+describe('Unit test - Greetings Handler', () => {
+	it('Should greet user including the username', () => {
+		const greeting = greetUser('Daniel');
+		expect(greeting).to.equal('¡Buenos días Daniel!');
+	});
```
Comment (Betisman) [inline test/unit/greetings.test.js]:
```suggestion
	});
	
```

File: test/testHelper/testHelper.js (hunk 2 above)
Comment (danielherrerohernando) [reply test/testHelper/testHelper.js]: Like it!

File: test/integration-e2e/ohceCliApp.test.js (hunk 1 above)
Comment (danielherrerohernando) [reply test/integration-e2e/ohceCliApp.test.js]: Like it!

Comment (danielherrerohernando) [reply test/integration-e2e/ohceCliApp.test.js]: Done

File: test/testHelper/testHelper.js (hunk 2 above)
Comment (danielherrerohernando) [reply test/testHelper/testHelper.js]: Done

File: test/unit/greetings.test.js (hunk 4 above)
Comment (danielherrerohernando) [reply test/unit/greetings.test.js]: Loving this too

**Pull request merged**